File: test-suite-python/src/test/python/micronaut/docs/metrics/exception/OutOfStockException.py
# tag::imports[]
from java.lang import RuntimeException
from micronaut.http import HttpResponse, HttpResponseProvider
# end::imports[]


# tag::class[]
class OutOfStockException(RuntimeException, HttpResponseProvider):

    def getResponse(self) -> HttpResponse:
        return HttpResponse.ok(0)
# end::class[]

File: test-suite/src/test/java/io/micronaut/docs/metrics/exception/OutOfStockException.java
package io.micronaut.docs.metrics.exception;

// tag::imports[]
import io.micronaut.http.HttpResponse;
import io.micronaut.http.HttpResponseProvider;
// end::imports[]

// tag::class[]
public class OutOfStockException extends RuntimeException implements HttpResponseProvider {

    @Override
    public HttpResponse<?> getResponse() {
        return HttpResponse.ok(0);
    }
}
// end::class[]
